pkg/common/windows.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// GetWindowsNodes returns Windows nodes, optionally filtering the list of nodes
+// with the given filter functions.
+package common
+
+import (
+	"fmt"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	// This taint is applied to nodes upgrading Calico Windows.
+	CalicoWindowsUpgradingTaint = &corev1.Taint{
+		Key:    CalicoWindowsUpgradeTaintKey,
+		Effect: corev1.TaintEffectNoSchedule,
```

Comment:
I think for now just preventing scheduling is enough. I chatted with Song about this and draining the node would be difficult to get right since the platform might also be actively evicting pods or setting taints on the node. We'll need to doc this as a limitation.

Reply:
I don't really follow - do we need to evict pods in order to do this upgrade gracefully, or can it be done gracefully just by preventing scheduling?

What is the limitation you're referring to?

Reply:
> I don't really follow - do we need to evict pods in order to do this upgrade gracefully, or can it be done gracefully just by preventing scheduling?

Not from my testing so far, no I don't believe evicting the pods is needed. 

> What is the limitation you're referring to?

Sorry, I was jumping the gun here. I don't see any limitations yet but we're ramping up the testing.